client/rest-high-level/src/main/java/org/elasticsearch/client/indices/PutIndexTemplateRequest.java

```diff
@@ -84,12 +85,11 @@ public PutIndexTemplateRequest(String name) {
     }
 
     @Override
-    public ActionRequestValidationException validate() {
-        ActionRequestValidationException validationException = null;
+    public Optional<ValidationException> validate() {
         if (indexPatterns == null || indexPatterns.size() == 0) {
-            validationException = addValidationError("index patterns are missing", validationException);
+            return Optional.of(ValidationException.withError("index patterns are missing"));
```

Comment:
I think that even tho this is like a builder and a request, we should put things that required (like in this case the indexPatterns) into a constructor for the builder. WDYT?

Then we move the validation to the constructor and remove this validate method altogether.